pkg/state/transaction_performer.go

```diff
@@ -344,6 +344,25 @@ func (tp *transactionPerformer) performInvokeScriptWithProofs(transaction proto.
 	return nil
 }
 
+func (tp *transactionPerformer) performEthereumTransactionWithProofs(transaction proto.Transaction, info *performerInfo) error {
+	ethTx, ok := transaction.(*proto.EthereumTransaction)
+	if !ok {
+		return errors.New("failed to convert interface to EthereumTransaction transaction")
+	}
+
+	switch ethTx.TxKind.(type) {
+	case *proto.EthereumTransferWavesTxKind, *proto.EthereumTransferAssetsErc20TxKind:
+		return nil
+	case *proto.EthereumInvokeScriptTxKind:
```

Comment:
Why not to check on certain kind, nil otherwise?

Reply:
Fixed